### PR TITLE
🐛 handle case where all assets report an error

### LINKED ISSUE #231: panics when invalid --policy filter is used
**Describe the bug**

as reported by @username-is-already-taken2

```
cnspec scan local --policy cnspec-policies/core/mondoo-linux-security.mql.yaml 
→ loaded configuration from /home/chris/.config/mondoo/mondoo.yml using source default
→ using service account credentials
→ discover related assets for 1 asset(s)
→ resolved assets resolved-assets=1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x63136ee]

goroutine 1 [running]:
go.mondoo.com/cnspec/policy.(*Bundle).FilterPolicies(0x0, {0xc000f541a0, 0x1, 0xc000fd1918?})
	/home/benr/actions-runner/_work/cnspec/cnspec/policy/bundle.go:222 +0x12e
go.mondoo.com/cnspec/policy/scan.(*LocalScanner).distributeJob(0xc000863320?, 0xc000fd1b58, {0x8ba2c28?, 0xc001611ec0}, {{0x0, 0x0}, {0xc00138c180, 0x32}, {0xc000582780, 0x19}, ...})
	/home/benr/actions-runner/_work/cnspec/cnspec/policy/scan/local_scanner.go:224 +0x5e8
go.mondoo.com/cnspec/policy/scan.(*LocalScanner).Run(0x8ba2bb8?, {0x8ba2c28?, 0xc001611e60?}, 0xc000fd1b58)
	/home/benr/actions-runner/_work/cnspec/cnspec/policy/scan/local_scanner.go:142 +0x15b
go.mondoo.com/cnspec/apps/cnspec/cmd.RunScan(0xc0010221b0)
	/home/benr/actions-runner/_work/cnspec/cnspec/apps/cnspec/cmd/scan.go:539 +0x35f
go.mondoo.com/cnspec/apps/cnspec/cmd.glob..func18(0x0?, {0xc0000ffb40?, 0x0?, 0x0?}, 0x0?, 0x0?)
	/home/benr/actions-runner/_work/cnspec/cnspec/apps/cnspec/cmd/scan.go:348 +0xd9
go.mondoo.com/cnquery/apps/cnquery/cmd/builder.localProviderCmd.func1(0xc001400000?, {0xc0000ffb40?, 0x2?, 0x2?})
	/home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20221207043121-7255f8d8fc78/apps/cnquery/cmd/builder/builder.go:193 +0x2b
github.com/spf13/cobra.(*Command).execute(0xc001400000, {0xc0000ffb20, 0x2, 0x2})
	/home/benr/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:920 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0xc192f60)
	/home/benr/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	/home/benr/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
go.mondoo.com/cnspec/apps/cnspec/cmd.Execute()
	/home/benr/actions-runner/_work/cnspec/cnspec/apps/cnspec/cmd/root.go:76 +0x25
main.main()
	/home/benr/actions-runner/_work/cnspec/cnspec/apps/cnspec/cnspec.go:6 +0x17
```

**To Reproduce**

Download latest cnspec and provide an invalid `--policy` filter

**Expected behavior**

- detect that the provided policy is not available and display a warning
- if the user provides an argument with `.mql.yaml` at the end, display warning that `--policy-bundle` is probably the right argument

**Screenshots or CLI Output**

n/a

**Desktop (please complete the following information):**
- OS: [e.g. macOS]
- OS Version: [e.g. 13.0]
- Browser if applicable [e.g. Chrome, Firefox]
- Browser Version: [e.g. 106]

**Additional context**

n/a

### PR DESCRIPTION
**Pull request by chris-rock**
fixes #231

- print warning if the `--policy` flag looks like a bundle
- renders asset errors when all assets have an error

Screenshot 2022-12-12 at 12 19 58

### CODE REVIEW

File: policy/scan/local_scanner.go [hunk 1]
```diff
@@ -568,6 +569,34 @@ func (s *localAssetScanner) run() (*AssetReport, error) {
...
+func noPolicyErr(availablePolicies []string, filter []string) error {
+	var sb strings.Builder
+	sb.WriteString("bundle doesn't contain any policies\n")
+	sb.WriteString("\n")
+
+	if len(availablePolicies) > 0 {
+		sb.WriteString("The following policies are available:\n")
+		for i := range availablePolicies {
+			policyMrn := availablePolicies[i]
+			sb.WriteString("- " + policyMrn)
```
Comment (jaym) [inline policy/scan/local_scanner.go]: just double checking that you didn't want a newline here

**Review by jaym — APPROVED** [review]

File: policy/scan/local_scanner.go (hunk 1 above)
Comment (chris-rock) [reply policy/scan/local_scanner.go]: Great catch, I'll add the newline there

File: policy/scan/local_scanner.go [hunk 2]
```diff
@@ -568,6 +569,34 @@ func (s *localAssetScanner) run() (*AssetReport, error) {
...
+		sb.WriteString("\n")
+	} else {
+		sb.WriteString("The policy bundle for the asset does not contain any policies\n\n")
+	}
+
+	if len(filter) > 0 {
+		sb.WriteString("User selected policies that are allowed to run:\n")
+		for i := range filter {
+			policyMrn := filter[i]
+			sb.WriteString("- " + policyMrn)
```
Comment (jaym) [inline policy/scan/local_scanner.go]: i think you might also want a newline here?

Comment (chris-rock) [reply policy/scan/local_scanner.go]: Indeed

**Pull request merged**